File: FizzBuzz/FizzBuzz.m
// Fizzbuzz solution in Obj-C for CodeEval.com by Steven A. Dunn

#import <Foundation/Foundation.h>
#import <stdio.h>

const int LINE_SIZE = 100;

NSString* fizzBuzz(int, int, int);

int main(int argc, char* argv[])
{
  NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
  FILE* inputFile = fopen(argv[1], "r");
  char rawLine[LINE_SIZE];
  if (inputFile != NULL)
  {
    while (fgets(rawLine, LINE_SIZE, inputFile) != NULL)
    {
      NSString* line = [NSString stringWithFormat:@"%s", rawLine];
      NSArray* tokens = [line componentsSeparatedByString: @" "];

      int A = [[tokens objectAtIndex:0] integerValue];
      int B = [[tokens objectAtIndex:1] integerValue];
      int N = [[tokens objectAtIndex:2] integerValue];

      NSString* result = fizzBuzz(A, B, N);
      puts([result UTF8String]);
    }
    fclose(inputFile);
  }
  [pool drain];
  return 0;
}

NSString* fizzBuzz(int A, int B, int N)
{
  NSString* result = [[NSString alloc] init];
  int i;
  for(i = 1; i <= N; ++i)
  {
    if (i % A == 0 && i % B == 0) 
      result = [result stringByAppendingString:@"FB"]; 
    else if (i % A == 0)
      result = [result stringByAppendingString:@"F"];
    else if (i % B == 0)
      result = [result stringByAppendingString:@"B"];
    else
      result = [result stringByAppendingFormat:@"%d", i];

    result = [result stringByAppendingString:@" "];
  }
  return result;
}

File: FizzBuzz/cpp/FizzBuzz.cpp
// Fizz Buzz program written in C++ for CodeEval.com by Steven A. Dunn

#include <fstream>
#include <iostream>
#include <sstream>
#include <stdio.h>
#include <string>

using std::cout;
using std::endl;
using std::ifstream;
using std::ofstream;
using std::ostringstream;
using std::string;
using std::stringstream;

string fizzBuzz(int a, int b, int n);

int main(int argc, char* argv[])
{
  ifstream inputFile(argv[1]);

  string result, line;
  stringstream ssline;

  int A, B, N = 0;

  if(inputFile)
  {
    while(getline(inputFile, line))
    {
      ssline << line;
      ssline >> A;
      ssline >> B;
      ssline >> N;
   
      result = fizzBuzz(A, B, N);
 
      ssline.clear();
      cout << result << endl;
    }   
  }
  inputFile.close();

  return 0;
}

string fizzBuzz(int a, int b, int n)
{
  string result;
  for(int i = 1; i <= n; i++)
  {
    if (i%a == 0 && i%b == 0) { result.append("FB"); }
    else if (i%a == 0) { result.append("F"); }
    else if (i%b == 0) { result.append("B"); }
    else 
    { 
      ostringstream oss;
      oss << i;
      result.append(oss.str());
      oss.flush(); 
    }
    result.append(" ");
  }

  int sz = result.size();  
  result.resize(sz-1);

  return result;
}